src/app/api/analytics/background_task.py

```diff
@@ -129,10 +138,88 @@ def run():
         }
 
         oer_search_store.append(
-            SearchStore(node_id=row.id, missing_materials=missing_materials)
+            SearchStore(node_id=collection.id, missing_materials=missing_materials)
         )
 
+        materials = [
+            PendingMaterials(**build_pending_materials(collection))
```

Comment:
why not return a PendingMaterials object from the function directly instead of a dictionary that then gets unpacked here?

Reply:
Good point, I will refactor this.

Reply:
I added it as todo to https://github.com/openeduhub/metaqs-main/issues/91